app/rust/src/parser/c32.rs

```diff
@@ -64,55 +66,75 @@ fn double_sha256_checksum(data: &mut [u8; SHA256_LEN]) {
 fn double_sha256_checksum(data: &mut [u8; SHA256_LEN]) {
     let mut output = [0u8; SHA256_LEN];
     // safe to unwrap as we are passing the right len
-    sha256(&data[..21], &mut output[..]).unwrap();
-    data.copy_from_slice(output.as_ref());
+    sha256(&data[..21], &mut output[..]).apdu_unwrap();
+    data.iter_mut()
+        .zip(output.as_ref().iter())
+        .for_each(|(d, o)| *d = *o);
     // safe to unwrap as we are passing the right len
-    sha256(&data[..], &mut output).unwrap();
-    data[20..24].copy_from_slice(&output[..4])
+    sha256(&data[..], &mut output).apdu_unwrap();
+    if let Some(d) = data.get_mut(20..24) {
```

Comment:
I think this may actually make 2 branches still, since in the `None` case then nothing is to be done...
`.get_mut(20..24).apdu_unwrap()` is what you need I think... 

As for the length check of copy_from_slice... I think it should be okay, but I'm way less confident compared to the above, so maybe it's better to help the compiler here